### ISSUE TITLE
Why does client send a <Description: Handshake Failure (40)>?

### ISSUE THREAD

**Issue by NannanMao** [openssl/openssl#21149]
client ->(ClientHello) server: OK
server ->(ServerHello) client: OK
server ->(Certificate, Server Hello Done) client: OK
client ->(Alert (Level: Fatal, Description: Handshake Failure)) server: Fatal Error

By the way, I use libcurl to do a HTTPS transport, and I add <CURLOPT_SSL_VERIFYHOST> and <CURLOPT_SSL_VERIFYPEER> to disable verifying the certificate of the server, (I am sure the verifying is closed, because I can connect to my own HTTPS server which uses a self signed certificate), but I still receive this error: curl Error 35, SSL connect failed when I connect to the real server.

**Client Hello**
```
Transport Layer Security
    TLSv1.2 Record Layer: Handshake Protocol: Client Hello
        Content Type: Handshake (22)
        Version: TLS 1.0 (0x0301)
        Length: 512
        Handshake Protocol: Client Hello
            Handshake Type: Client Hello (1)
            Length: 508
            Version: TLS 1.2 (0x0303)
            Random: 4e5aed46586b72b0ab647eaf740b564ef44fb71440af9ee1c4f399e4433e1229
                GMT Unix Time: Aug 29, 2011 09:37:10.000000000 中国标准时间
                Random Bytes: 586b72b0ab647eaf740b564ef44fb71440af9ee1c4f399e4433e1229
            Session ID Length: 32
            Session ID: 06bc0d9abf5efb1767db23c8530e33944b8d08e522f81f61038613d3e6bef0e2
            Cipher Suites Length: 62
            Cipher Suites (31 suites)
                Cipher Suite: TLS_AES_256_GCM_SHA384 (0x1302)
                Cipher Suite: TLS_CHACHA20_POLY1305_SHA256 (0x1303)
                Cipher Suite: TLS_AES_128_GCM_SHA256 (0x1301)
                Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 (0xc02c)
                Cipher Suite: TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (0xc030)
                Cipher Suite: TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 (0x009f)
                Cipher Suite: TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 (0xcca9)
                Cipher Suite: TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 (0xcca8)
                Cipher Suite: TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256 (0xccaa)
                Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 (0xc02b)
                Cipher Suite: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)
                Cipher Suite: TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 (0x009e)
                Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384 (0xc024)
                Cipher Suite: TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (0xc028)
                Cipher Suite: TLS_DHE_RSA_WITH_AES_256_CBC_SHA256 (0x006b)
                Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256 (0xc023)
                Cipher Suite: TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (0xc027)
                Cipher Suite: TLS_DHE_RSA_WITH_AES_128_CBC_SHA256 (0x0067)
                Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA (0xc00a)
                Cipher Suite: TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (0xc014)
                Cipher Suite: TLS_DHE_RSA_WITH_AES_256_CBC_SHA (0x0039)
                Cipher Suite: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA (0xc009)
                Cipher Suite: TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (0xc013)
                Cipher Suite: TLS_DHE_RSA_WITH_AES_128_CBC_SHA (0x0033)
                Cipher Suite: TLS_RSA_WITH_AES_256_GCM_SHA384 (0x009d)
                Cipher Suite: TLS_RSA_WITH_AES_128_GCM_SHA256 (0x009c)
                Cipher Suite: TLS_RSA_WITH_AES_256_CBC_SHA256 (0x003d)
                Cipher Suite: TLS_RSA_WITH_AES_128_CBC_SHA256 (0x003c)
                Cipher Suite: TLS_RSA_WITH_AES_256_CBC_SHA (0x0035)
                Cipher Suite: TLS_RSA_WITH_AES_128_CBC_SHA (0x002f)
                Cipher Suite: TLS_EMPTY_RENEGOTIATION_INFO_SCSV (0x00ff)
            Compression Methods Length: 1
            Compression Methods (1 method)
                Compression Method: null (0)
            Extensions Length: 373
            Extension: server_name (len=32)
                Type: server_name (0)
                Length: 32
                Server Name Indication extension
                    Server Name list length: 30
                    Server Name Type: host_name (0)
                    Server Name length: 27
                    Server Name: umapmgs.test.abchina.com.cn
            Extension: ec_point_formats (len=4)
                Type: ec_point_formats (11)
                Length: 4
                EC point formats Length: 3
                Elliptic curves point formats (3)
                    EC point format: uncompressed (0)
                    EC point format: ansiX962_compressed_prime (1)
                    EC point format: ansiX962_compressed_char2 (2)
            Extension: supported_groups (len=22)
                Type: supported_groups (10)
                Length: 22
                Supported Groups List Length: 20
                Supported Groups (10 groups)
                    Supported Group: x25519 (0x001d)
                    Supported Group: secp256r1 (0x0017)
                    Supported Group: x448 (0x001e)
                    Supported Group: secp521r1 (0x0019)
                    Supported Group: secp384r1 (0x0018)
                    Supported Group: ffdhe2048 (0x0100)
                    Supported Group: ffdhe3072 (0x0101)
                    Supported Group: ffdhe4096 (0x0102)
                    Supported Group: ffdhe6144 (0x0103)
                    Supported Group: ffdhe8192 (0x0104)
            Extension: next_protocol_negotiation (len=0)
                Type: next_protocol_negotiation (13172)
                Length: 0
            Extension: application_layer_protocol_negotiation (len=14)
                Type: application_layer_protocol_negotiation (16)
                Length: 14
                ALPN Extension Length: 12
                ALPN Protocol
                    ALPN string length: 2
                    ALPN Next Protocol: h2
                    ALPN string length: 8
                    ALPN Next Protocol: http/1.1
            Extension: encrypt_then_mac (len=0)
                Type: encrypt_then_mac (22)
                Length: 0
            Extension: extended_master_secret (len=0)
                Type: extended_master_secret (23)
                Length: 0
            Extension: post_handshake_auth (len=0)
                Type: post_handshake_auth (49)
                Length: 0
            Extension: signature_algorithms (len=42)
                Type: signature_algorithms (13)
                Length: 42
                Signature Hash Algorithms Length: 40
                Signature Hash Algorithms (20 algorithms)
                    Signature Algorithm: ecdsa_secp256r1_sha256 (0x0403)
                        Signature Hash Algorithm Hash: SHA256 (4)
                        Signature Hash Algorithm Signature: ECDSA (3)
                    Signature Algorithm: ecdsa_secp384r1_sha384 (0x0503)
                        Signature Hash Algorithm Hash: SHA384 (5)
                        Signature Hash Algorithm Signature: ECDSA (3)
                    Signature Algorithm: ecdsa_secp521r1_sha512 (0x0603)
                        Signature Hash Algorithm Hash: SHA512 (6)
                        Signature Hash Algorithm Signature: ECDSA (3)
                    Signature Algorithm: ed25519 (0x0807)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: Unknown (7)
                    Signature Algorithm: ed448 (0x0808)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: Unknown (8)
                    Signature Algorithm: rsa_pss_pss_sha256 (0x0809)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: Unknown (9)
                    Signature Algorithm: rsa_pss_pss_sha384 (0x080a)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: Unknown (10)
                    Signature Algorithm: rsa_pss_pss_sha512 (0x080b)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: Unknown (11)
                    Signature Algorithm: rsa_pss_rsae_sha256 (0x0804)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: SM2 (4)
                    Signature Algorithm: rsa_pss_rsae_sha384 (0x0805)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: Unknown (5)
                    Signature Algorithm: rsa_pss_rsae_sha512 (0x0806)
                        Signature Hash Algorithm Hash: Unknown (8)
                        Signature Hash Algorithm Signature: Unknown (6)
                    Signature Algorithm: rsa_pkcs1_sha256 (0x0401)
                        Signature Hash Algorithm Hash: SHA256 (4)
                        Signature Hash Algorithm Signature: RSA (1)
                    Signature Algorithm: rsa_pkcs1_sha384 (0x0501)
                        Signature Hash Algorithm Hash: SHA384 (5)
                        Signature Hash Algorithm Signature: RSA (1)
                    Signature Algorithm: rsa_pkcs1_sha512 (0x0601)
                        Signature Hash Algorithm Hash: SHA512 (6)
                        Signature Hash Algorithm Signature: RSA (1)
                    Signature Algorithm: SHA224 ECDSA (0x0303)
                        Signature Hash Algorithm Hash: SHA224 (3)
                        Signature Hash Algorithm Signature: ECDSA (3)
                    Signature Algorithm: SHA224 RSA (0x0301)
                        Signature Hash Algorithm Hash: SHA224 (3)
                        Signature Hash Algorithm Signature: RSA (1)
                    Signature Algorithm: SHA224 DSA (0x0302)
                        Signature Hash Algorithm Hash: SHA224 (3)
                        Signature Hash Algorithm Signature: DSA (2)
                    Signature Algorithm: SHA256 DSA (0x0402)
                        Signature Hash Algorithm Hash: SHA256 (4)
                        Signature Hash Algorithm Signature: DSA (2)
                    Signature Algorithm: SHA384 DSA (0x0502)
                        Signature Hash Algorithm Hash: SHA384 (5)
                        Signature Hash Algorithm Signature: DSA (2)
                    Signature Algorithm: SHA512 DSA (0x0602)
                        Signature Hash Algorithm Hash: SHA512 (6)
                        Signature Hash Algorithm Signature: DSA (2)
            Extension: supported_versions (len=9)
                Type: supported_versions (43)
                Length: 9
                Supported Versions length: 8
                Supported Version: TLS 1.3 (0x0304)
                Supported Version: TLS 1.2 (0x0303)
                Supported Version: TLS 1.1 (0x0302)
                Supported Version: TLS 1.0 (0x0301)
            Extension: psk_key_exchange_modes (len=2)
                Type: psk_key_exchange_modes (45)
                Length: 2
                PSK Key Exchange Modes Length: 1
                PSK Key Exchange Mode: PSK with (EC)DHE key establishment (psk_dhe_ke) (1)
            Extension: key_share (len=38)
                Type: key_share (51)
                Length: 38
                Key Share extension
                    Client Key Share Length: 36
                    Key Share Entry: Group: x25519, Key Exchange length: 32
                        Group: x25519 (29)
                        Key Exchange Length: 32
                        Key Exchange: b6db69377bd462abfa5a2fc10e4e553032314d2605a792b80641cd474934a46a
            Extension: padding (len=158)
                Type: padding (21)
                Length: 158
                Padding Data: 000000000000000000000000000000000000000000000000000000000000000000000000…
            [JA3 Fullstring [truncated]: 771,4866-4867-4865-49196-49200-159-52393-52392-52394-49195-49199-158-49188-49192-107-49187-49191-103-49162-49172-57-49161-49171-51-157-156-61-60-53-47-255,0-11-10-13172-16-22-23-49-13-43-45-51-21,29-23-30-25-24]
            [JA3: 4ea056e63b7910cbf543f0c095064dfe]
```

**Server Hello**
```
Transport Layer Security
    TLSv1.2 Record Layer: Handshake Protocol: Server Hello
        Content Type: Handshake (22)
        Version: TLS 1.2 (0x0303)
        Length: 80
        Handshake Protocol: Server Hello
            Handshake Type: Server Hello (2)
            Length: 76
            Version: TLS 1.2 (0x0303)
            Random: 64807451400836b3843537a17149e4141024b3a5cd3c79ba87570b7f1f4e7c42
                GMT Unix Time: Jun  7, 2023 20:13:05.000000000 中国标准时间
                Random Bytes: 400836b3843537a17149e4141024b3a5cd3c79ba87570b7f1f4e7c42
            Session ID Length: 32
            Session ID: 07df177c18dbdc4db4284b4da1ec7b244ecc0104b78ad16953753f717c8ea909
            Cipher Suite: TLS_RSA_WITH_AES_256_GCM_SHA384 (0x009d)
            Compression Method: null (0)
            Extensions Length: 4
            Extension: server_name (len=0)
                Type: server_name (0)
                Length: 0
            [JA3S Fullstring: 771,157,0]
            [JA3S: f2f7d61373f825f7fd48cdac3370f9e7]
```

**Certificate**
```
Transport Layer Security
    TLSv1.2 Record Layer: Handshake Protocol: Certificate
        Content Type: Handshake (22)
        Version: TLS 1.2 (0x0303)
        Length: 3047
        Handshake Protocol: Certificate
            Handshake Type: Certificate (11)
            Length: 3043
            Certificates Length: 3040
            Certificates (3040 bytes)
                Certificate Length: 1728
                Certificate: 308206bc308205a4a00302010202100e4d0ad293e4f82adad349526beae839300d06092a… (id-at-commonName=*.test.abchina.com.cn,id-at-organizationName=Agricultural Bank Of China Limited,id-at-stateOrProvinceName=北京市,id-at-countryNam
                    signedCertificate
                        version: v3 (2)
                        serialNumber: 0x0e4d0ad293e4f82adad349526beae839
                        signature (sha256WithRSAEncryption)
                            Algorithm Id: 1.2.840.113549.1.1.11 (sha256WithRSAEncryption)
                        issuer: rdnSequence (0)
                            rdnSequence: 4 items (id-at-commonName=Secure Site CA G2,id-at-organizationalUnitName=www.digicert.com,id-at-organizationName=DigiCert Inc,id-at-countryName=US)
                                RDNSequence item: 1 item (id-at-countryName=US)
                                    RelativeDistinguishedName item (id-at-countryName=US)
                                        Object Id: 2.5.4.6 (id-at-countryName)
                                        CountryName: US
                                RDNSequence item: 1 item (id-at-organizationName=DigiCert Inc)
                                    RelativeDistinguishedName item (id-at-organizationName=DigiCert Inc)
                                        Object Id: 2.5.4.10 (id-at-organizationName)
                                        DirectoryString: printableString (1)
                                            printableString: DigiCert Inc
                                RDNSequence item: 1 item (id-at-organizationalUnitName=www.digicert.com)
                                    RelativeDistinguishedName item (id-at-organizationalUnitName=www.digicert.com)
                                        Object Id: 2.5.4.11 (id-at-organizationalUnitName)
                                        DirectoryString: printableString (1)
                                            printableString: www.digicert.com
                                RDNSequence item: 1 item (id-at-commonName=Secure Site CA G2)
                                    RelativeDistinguishedName item (id-at-commonName=Secure Site CA G2)
                                        Object Id: 2.5.4.3 (id-at-commonName)
                                        DirectoryString: printableString (1)
                                            printableString: Secure Site CA G2
                        validity
                            notBefore: utcTime (0)
                                utcTime: 2023-04-28 00:00:00 (UTC)
                            notAfter: utcTime (0)
                                utcTime: 2024-05-01 23:59:59 (UTC)
                        subject: rdnSequence (0)
                            rdnSequence: 4 items (id-at-commonName=*.test.abchina.com.cn,id-at-organizationName=Agricultural Bank Of China Limited,id-at-stateOrProvinceName=北京市,id-at-countryName=CN)
                                RDNSequence item: 1 item (id-at-countryName=CN)
                                    RelativeDistinguishedName item (id-at-countryName=CN)
                                        Object Id: 2.5.4.6 (id-at-countryName)
                                        CountryName: CN
                                RDNSequence item: 1 item (id-at-stateOrProvinceName=北京市)
                                    RelativeDistinguishedName item (id-at-stateOrProvinceName=北京市)
                                        Object Id: 2.5.4.8 (id-at-stateOrProvinceName)
                                        DirectoryString: uTF8String (4)
                                            uTF8String: 北京市
                                RDNSequence item: 1 item (id-at-organizationName=Agricultural Bank Of China Limited)
                                    RelativeDistinguishedName item (id-at-organizationName=Agricultural Bank Of China Limited)
                                        Object Id: 2.5.4.10 (id-at-organizationName)
                                        DirectoryString: printableString (1)
                                            printableString: Agricultural Bank Of China Limited
                                RDNSequence item: 1 item (id-at-commonName=*.test.abchina.com.cn)
                                    RelativeDistinguishedName item (id-at-commonName=*.test.abchina.com.cn)
                                        Object Id: 2.5.4.3 (id-at-commonName)
                                        DirectoryString: uTF8String (4)
                                            uTF8String: *.test.abchina.com.cn
                        subjectPublicKeyInfo
                            algorithm (rsaEncryption)
                                Algorithm Id: 1.2.840.113549.1.1.1 (rsaEncryption)
                            subjectPublicKey: 3082010a0282010100a877ed7f86a76db49d146d50f096ca3df896eda1bf3392f7cdf5c6…
                                modulus: 0x00a877ed7f86a76db49d146d50f096ca3df896eda1bf3392f7cdf5c62bc3ff49e69b66d4…
                                publicExponent: 65537
                        extensions: 10 items
                            Extension (id-ce-authorityKeyIdentifier)
                                Extension Id: 2.5.29.35 (id-ce-authorityKeyIdentifier)
                                AuthorityKeyIdentifier
                                    keyIdentifier: c4117e884086c241bf65f31ae1b45340a3abec7d
                            Extension (id-ce-subjectKeyIdentifier)
                                Extension Id: 2.5.29.14 (id-ce-subjectKeyIdentifier)
                                SubjectKeyIdentifier: 3c084d3d3ddd85e26129346292b4f2bd75c33dc3
                            Extension (id-ce-subjectAltName)
                                Extension Id: 2.5.29.17 (id-ce-subjectAltName)
                                GeneralNames: 3 items
                                    GeneralName: dNSName (2)
                                        dNSName: *.test.abchina.com.cn
                                    GeneralName: dNSName (2)
                                        dNSName: *.test.abchina.com
                                    GeneralName: dNSName (2)
                                        dNSName: *.test.95599.cn
                            Extension (id-ce-keyUsage)
                                Extension Id: 2.5.29.15 (id-ce-keyUsage)
                                critical: True
                                Padding: 5
                                KeyUsage: a0
                                    1... .... = digitalSignature: True
                                    .0.. .... = contentCommitment: False
                                    ..1. .... = keyEncipherment: True
                                    ...0 .... = dataEncipherment: False
                                    .... 0... = keyAgreement: False
                                    .... .0.. = keyCertSign: False
                                    .... ..0. = cRLSign: False
                                    .... ...0 = encipherOnly: False
                                    0... .... = decipherOnly: False
                            Extension (id-ce-extKeyUsage)
                                Extension Id: 2.5.29.37 (id-ce-extKeyUsage)
                                KeyPurposeIDs: 2 items
                                    KeyPurposeId: 1.3.6.1.5.5.7.3.1 (id-kp-serverAuth)
                                    KeyPurposeId: 1.3.6.1.5.5.7.3.2 (id-kp-clientAuth)
                            Extension (id-ce-cRLDistributionPoints)
                                Extension Id: 2.5.29.31 (id-ce-cRLDistributionPoints)
                                CRLDistPointsSyntax: 2 items
                                    DistributionPoint
                                        distributionPoint: fullName (0)
                                            fullName: 1 item
                                                GeneralName: uniformResourceIdentifier (6)
                                                    uniformResourceIdentifier: http://crl3.digicert.com/SecureSiteCAG2.crl
                                    DistributionPoint
                                        distributionPoint: fullName (0)
                                            fullName: 1 item
                                                GeneralName: uniformResourceIdentifier (6)
                                                    uniformResourceIdentifier: http://crl4.digicert.com/SecureSiteCAG2.crl
                            Extension (id-ce-certificatePolicies)
                                Extension Id: 2.5.29.32 (id-ce-certificatePolicies)
                                CertificatePoliciesSyntax: 1 item
                                    PolicyInformation
                                        policyIdentifier: 2.23.140.1.2.2 (joint-iso-itu-t.23.140.1.2.2)
                                        policyQualifiers: 1 item
                                            PolicyQualifierInfo
                                                Id: 1.3.6.1.5.5.7.2.1 (id-qt-cps)
                                                DirectoryString: http://www.digicert.com/CPS
                            Extension (id-pe-authorityInfoAccess)
                                Extension Id: 1.3.6.1.5.5.7.1.1 (id-pe-authorityInfoAccess)
                                AuthorityInfoAccessSyntax: 2 items
                                    AccessDescription
                                        accessMethod: 1.3.6.1.5.5.7.48.1 (id-ad-ocsp)
                                        accessLocation: 6
                                            uniformResourceIdentifier: http://ocsp.dcocsp.cn
                                    AccessDescription
                                        accessMethod: 1.3.6.1.5.5.7.48.2 (id-ad-caIssuers)
                                        accessLocation: 6
                                            uniformResourceIdentifier: http://crl.digicert-cn.com/SecureSiteCAG2.crt
                            Extension (id-ce-basicConstraints)
                                Extension Id: 2.5.29.19 (id-ce-basicConstraints)
                                BasicConstraintsSyntax [0 length]
                            Extension (SignedCertificateTimestampList)
                                Extension Id: 1.3.6.1.4.1.11129.2.4.2 (SignedCertificateTimestampList)
                                Serialized SCT List Length: 361
                                Signed Certificate Timestamp (Google 'Xenon2024' log)
                                    Serialized SCT Length: 118
                                    SCT Version: 0
                                    Log ID: 76ff883f0ab6fb9551c261ccf587ba34b4a4cdbb29dc68420a9fe6674c5a3a74
                                    Timestamp: Apr 28, 2023 02:25:55.386000000 UTC
                                    Extensions length: 0
                                    Signature Algorithm: ecdsa_secp256r1_sha256 (0x0403)
                                        Signature Hash Algorithm Hash: SHA256 (4)
                                        Signature Hash Algorithm Signature: ECDSA (3)
                                    Signature Length: 71
                                    Signature: 3045022100e3ffb27428fe3629fedfa2bbe9206dc99c26b7ad6c8789d6554dd4ab36e7c6…
                                Signed Certificate Timestamp (DigiCert Nessie2024 Log)
                                    Serialized SCT Length: 119
                                    SCT Version: 0
                                    Log ID: 73d99e891b4c9678a0207d479de6b2c61cd0515e71192a8c6b80107ac17772b5
                                    Timestamp: Apr 28, 2023 02:25:55.412000000 UTC
                                    Extensions length: 0
                                    Signature Algorithm: ecdsa_secp256r1_sha256 (0x0403)
                                        Signature Hash Algorithm Hash: SHA256 (4)
                                        Signature Hash Algorithm Signature: ECDSA (3)
                                    Signature Length: 72
                                    Signature: 3046022100858d164a4fa23d7d6410420eda1b36cad7ad2825e84e285bca394b0318979c…
                                Signed Certificate Timestamp (DigiCert Yeti2024 Log)
                                    Serialized SCT Length: 118
                                    SCT Version: 0
                                    Log ID: 48b0e36bdaa647340fe56a02fa9d30eb1c5201cb56dd2c81d9bbbfab39d88473
                                    Timestamp: Apr 28, 2023 02:25:55.350000000 UTC
                                    Extensions length: 0
                                    Signature Algorithm: ecdsa_secp256r1_sha256 (0x0403)
                                        Signature Hash Algorithm Hash: SHA256 (4)
                                        Signature Hash Algorithm Signature: ECDSA (3)
                                    Signature Length: 71
                                    Signature: 30450220336c131bde3d21c9f08ceb06aa9fc954fed44f0a01526c1baa91cfeba68bbc90…
                    algorithmIdentifier (sha256WithRSAEncryption)
                        Algorithm Id: 1.2.840.113549.1.1.11 (sha256WithRSAEncryption)
                    Padding: 0
                    encrypted: 285d4562487c570340b27cfca3561d7c6c2a22b0347f09fafa157ca6769db5e773155024…
                Certificate Length: 1306
                Certificate: 30820516308203fea0030201020210087e18fb8faa927ee8760a0d8982f512300d06092a… (id-at-commonName=Secure Site CA G2,id-at-organizationalUnitName=www.digicert.com,id-at-organizationName=DigiCert Inc,id-at-countryName=US)
                    signedCertificate
                        version: v3 (2)
                        serialNumber: 0x087e18fb8faa927ee8760a0d8982f512
                        signature (sha256WithRSAEncryption)
                            Algorithm Id: 1.2.840.113549.1.1.11 (sha256WithRSAEncryption)
                        issuer: rdnSequence (0)
                            rdnSequence: 4 items (id-at-commonName=DigiCert Global Root CA,id-at-organizationalUnitName=www.digicert.com,id-at-organizationName=DigiCert Inc,id-at-countryName=US)
                                RDNSequence item: 1 item (id-at-countryName=US)
                                    RelativeDistinguishedName item (id-at-countryName=US)
                                        Object Id: 2.5.4.6 (id-at-countryName)
                                        CountryName: US
                                RDNSequence item: 1 item (id-at-organizationName=DigiCert Inc)
                                    RelativeDistinguishedName item (id-at-organizationName=DigiCert Inc)
                                        Object Id: 2.5.4.10 (id-at-organizationName)
                                        DirectoryString: printableString (1)
                                            printableString: DigiCert Inc
                                RDNSequence item: 1 item (id-at-organizationalUnitName=www.digicert.com)
                                    RelativeDistinguishedName item (id-at-organizationalUnitName=www.digicert.com)
                                        Object Id: 2.5.4.11 (id-at-organizationalUnitName)
                                        DirectoryString: printableString (1)
                                            printableString: www.digicert.com
                                RDNSequence item: 1 item (id-at-commonName=DigiCert Global Root CA)
                                    RelativeDistinguishedName item (id-at-commonName=DigiCert Global Root CA)
                                        Object Id: 2.5.4.3 (id-at-commonName)
                                        DirectoryString: printableString (1)
                                            printableString: DigiCert Global Root CA
                        validity
                            notBefore: utcTime (0)
                                utcTime: 2019-06-20 12:21:35 (UTC)
                            notAfter: utcTime (0)
                                utcTime: 2029-06-20 12:21:35 (UTC)
                        subject: rdnSequence (0)
                            rdnSequence: 4 items (id-at-commonName=Secure Site CA G2,id-at-organizationalUnitName=www.digicert.com,id-at-organizationName=DigiCert Inc,id-at-countryName=US)
                                RDNSequence item: 1 item (id-at-countryName=US)
                                    RelativeDistinguishedName item (id-at-countryName=US)
                                        Object Id: 2.5.4.6 (id-at-countryName)
                                        CountryName: US
                                RDNSequence item: 1 item (id-at-organizationName=DigiCert Inc)
                                    RelativeDistinguishedName item (id-at-organizationName=DigiCert Inc)
                                        Object Id: 2.5.4.10 (id-at-organizationName)
                                        DirectoryString: printableString (1)
                                            printableString: DigiCert Inc
                                RDNSequence item: 1 item (id-at-organizationalUnitName=www.digicert.com)
                                    RelativeDistinguishedName item (id-at-organizationalUnitName=www.digicert.com)
                                        Object Id: 2.5.4.11 (id-at-organizationalUnitName)
                                        DirectoryString: printableString (1)
                                            printableString: www.digicert.com
                                RDNSequence item: 1 item (id-at-commonName=Secure Site CA G2)
                                    RelativeDistinguishedName item (id-at-commonName=Secure Site CA G2)
                                        Object Id: 2.5.4.3 (id-at-commonName)
                                        DirectoryString: printableString (1)
                                            printableString: Secure Site CA G2
                        subjectPublicKeyInfo
                            algorithm (rsaEncryption)
                                Algorithm Id: 1.2.840.113549.1.1.1 (rsaEncryption)
                            subjectPublicKey: 3082010a0282010100c7bb3dd377d1e92829034f147612845086476b6220eca3cd913b70…
                                modulus: 0x00c7bb3dd377d1e92829034f147612845086476b6220eca3cd913b70312d7ed3c612e504…
                                publicExponent: 65537
                        extensions: 8 items
                            Extension (id-ce-subjectKeyIdentifier)
                                Extension Id: 2.5.29.14 (id-ce-subjectKeyIdentifier)
                                SubjectKeyIdentifier: c4117e884086c241bf65f31ae1b45340a3abec7d
                            Extension (id-ce-authorityKeyIdentifier)
                                Extension Id: 2.5.29.35 (id-ce-authorityKeyIdentifier)
                                AuthorityKeyIdentifier
                                    keyIdentifier: 03de503556d14cbb66f0a3e21b1bc397b23dd155
                            Extension (id-ce-keyUsage)
                                Extension Id: 2.5.29.15 (id-ce-keyUsage)
                                critical: True
                                Padding: 1
                                KeyUsage: 86
                                    1... .... = digitalSignature: True
                                    .0.. .... = contentCommitment: False
                                    ..0. .... = keyEncipherment: False
                                    ...0 .... = dataEncipherment: False
                                    .... 0... = keyAgreement: False
                                    .... .1.. = keyCertSign: True
                                    .... ..1. = cRLSign: True
                                    .... ...0 = encipherOnly: False
                                    0... .... = decipherOnly: False
                            Extension (id-ce-extKeyUsage)
                                Extension Id: 2.5.29.37 (id-ce-extKeyUsage)
                                KeyPurposeIDs: 2 items
                                    KeyPurposeId: 1.3.6.1.5.5.7.3.1 (id-kp-serverAuth)
                                    KeyPurposeId: 1.3.6.1.5.5.7.3.2 (id-kp-clientAuth)
                            Extension (id-ce-basicConstraints)
                                Extension Id: 2.5.29.19 (id-ce-basicConstraints)
                                critical: True
                                BasicConstraintsSyntax
                                    cA: True
                            Extension (id-pe-authorityInfoAccess)
                                Extension Id: 1.3.6.1.5.5.7.1.1 (id-pe-authorityInfoAccess)
                                AuthorityInfoAccessSyntax: 1 item
                                    AccessDescription
                                        accessMethod: 1.3.6.1.5.5.7.48.1 (id-ad-ocsp)
                                        accessLocation: 6
                                            uniformResourceIdentifier: http://ocsp.dcocsp.cn
                            Extension (id-ce-cRLDistributionPoints)
                                Extension Id: 2.5.29.31 (id-ce-cRLDistributionPoints)
                                CRLDistPointsSyntax: 1 item
                                    DistributionPoint
                                        distributionPoint: fullName (0)
                                            fullName: 1 item
                                                GeneralName: uniformResourceIdentifier (6)
                                                    uniformResourceIdentifier: http://crl.digicert-cn.com/DigiCertGlobalRootCA.crl
                            Extension (id-ce-certificatePolicies)
                                Extension Id: 2.5.29.32 (id-ce-certificatePolicies)
                                CertificatePoliciesSyntax: 1 item
                                    PolicyInformation
                                        policyIdentifier: 2.5.29.32.0 (anyPolicy)
                                        policyQualifiers: 2 items
                                            PolicyQualifierInfo
                                                Id: 1.3.6.1.5.5.7.2.1 (id-qt-cps)
                                                DirectoryString: https://www.digicert.com/CPS
                                            PolicyQualifierInfo
                                                Id: 1.3.6.1.5.5.7.2.2 (id-qt-unotice)
                                                UserNotice
                                                    explicitText: utf8String (3)
                                                        utf8String: Any use of this Certificate constitutes acceptance of the Relying Party Agreement located at https://www.digicert.com/rpa-ua
                    algorithmIdentifier (sha256WithRSAEncryption)
                        Algorithm Id: 1.2.840.113549.1.1.11 (sha256WithRSAEncryption)
                    Padding: 0
                    encrypted: 13ef255b963022e891b079f881845155b2e6232a56c181fbe252179d08802de52c52459f…
```
**Server Hello Done**
```
Transport Layer Security
    TLSv1.2 Record Layer: Handshake Protocol: Server Hello Done
        Content Type: Handshake (22)
        Version: TLS 1.2 (0x0303)
        Length: 4
        Handshake Protocol: Server Hello Done
            Handshake Type: Server Hello Done (14)
            Length: 0
```

**Client Alert**
```
Transport Layer Security
    TLSv1.2 Record Layer: Alert (Level: Fatal, Description: Handshake Failure)
        Content Type: Alert (21)
        Version: TLS 1.2 (0x0303)
        Length: 2
        Alert Message
            Level: Fatal (2)
            Description: Handshake Failure (40)
```

**Comment by mattcaswell** [comment]
Do you get any error message from the client? Without that its quite difficult to say what the cause of this failure is. "Handshake failure" is quite a generic error.

A couple of things that look slightly odd about the server is that it does not seem to support "extended master secret", and it also does not seem to support "secure renegotiation". Neither of these would normally cause an OpenSSL client to fail though.

**Comment by Paulbo** [comment]
You can try adjusting the content of different extensions. I also encountered a similar issue. Initially, I thought it was due to the cipher and sigalgs, but after trying different settings, I found that the server-side groups did not support x448.

**Comment by nhorman** [comment]
marknig as inactive, due to no response from the reporter in more than a year.  Will be closed at the end of 3.4 dev cycle without further input
